onnxruntime/core/session/inference_session.cc

```diff
@@ -955,12 +1020,45 @@ class InferenceSession::Impl {
     return Status::OK();
   }
 
+  Status ValidateLevel(unsigned int level) const {
+    auto t_level = (unsigned int)std::pow(2, level);
+    if (t_level < TransformerLevel::MaxTransformerLevel) {
+      return Status::OK();
+    }
+    return Status(ONNXRUNTIME, FAIL, "Unsupported level " + level);
+  }
+
+  void SetTransformerContext(uint32_t& transformers_enabled, std::vector<TransformerLevel>& all_levels, const uint32_t& level) {
+    ORT_ENFORCE(ValidateLevel(level).IsOK(),
+                "Unsupported transformer level specified", level);
+
```

Comment:
Error messages should ideally make it easy for the user to fix the problem. In this case, directly telling them what is valid would be good.